slam-libraries/viam-cartographer/src/mapping/map_builder.cc

```diff
@@ -96,6 +96,20 @@ cartographer::sensor::TimedPointCloudData MapBuilder::GetDataFromFile(
 
 // TODO: There might still be a lot of room to improve accuracy & speed.
 // Might be worth investigating in the future.
+cartographer::transform::Rigid3d MapBuilder::GetLastGlobalPose(
+    int trajectory_id) {
+    if (GetLocalSlamResultPoses().size() > 0) {
```

Comment:
Pull out `GetLocalSlamResultPoses()` and save the result in `local_poses`. Reuse `local_poses`, that way you don't have to call `GetLocalSlamResultPoses()` twice (which might be doing a lot of work behind the scenes)